src/main/java/frc/robot/subsystems/drivetrain/SwerveDrive.java

```diff
@@ -263,11 +246,17 @@ public void periodic() {
         loggerInputs.statorCurrent =
                 modules[0].getStatorCurrent() + modules[1].getStatorCurrent() + modules[2].getStatorCurrent() + modules[3].getStatorCurrent();
 
-        loggerInputs.pitch = getPitch();
-        loggerInputs.rawYaw = getRawYaw();
-        loggerInputs.yaw = getYaw();
+        loggerInputs.rawYaw = gyro.getRawYaw();
+        loggerInputs.yaw = gyro.getYaw();
+        loggerInputs.pitch = gyro.getPitch();
         gyro.updateInputs(loggerInputs);
 
+//        desiredModuleStates = Utils.arrayToSwerveModuleStates(loggerInputs.desiredModuleStates);
+        SwerveDriveKinematics.desaturateWheelSpeeds(Utils.arrayToSwerveModuleStates(loggerInputs.desiredModuleStates), SwerveConstants.MAX_X_Y_VELOCITY); //TODO: may not work
+        for (int i = 0; i < modules.length; i++) {
+            modules[i].setModuleState(Utils.arrayToSwerveModuleStates(loggerInputs.desiredModuleStates)[i]);
+        }
```

Comment:
Is the conversion back and forth to enable logging it with AdvantageKit? If so, I believe this is unnecessary because AdvantageKit directly supports logging arrays of `SwerveModuleState`: https://www.chiefdelphi.com/t/advantagekit-2023-beta-log-replay-framework-now-with-wpilog-nt4-integration/417423/45?u=dan
You might need to upgrade your AdvantageKit's version to  [`v3.0.0-beta-1`](https://github.com/Mechanical-Advantage/AdvantageKit/releases/tag/v3.0.0-beta-1). To do this you'll also need to upgrade WPILib's version to the 2024 beta.